### PR TITLE
Add HTML Doxygen docs

### PR DESCRIPTION
**Pull request by alexmwu**
Add Doxyfile that generates doxygen web pages.

Ideas wanted for how to build and display Doxygen pages (add to Travis build probably, but where should we push the pages out?)

### CODE REVIEW

**Comment by dthain** [comment]
Auto-generated API docs are certainly fine, but the project also needs documentation at a higher level.
What is NunyaOS?  What are the design principles?  How do the components work together?

File: src/cmd_line.h [hunk 1]
```diff
@@ -8,8 +8,8 @@ See the file LICENSE for details.
 #define CMD_LINE_H
 
 /**
-* @brief Attempt to run the command given with its arguments
-* @detail Splits the line on whitespace, taking the first as the command and the rest as arguments. Will print "command not found" message if command is not found.
+* Attempt to run the command given with its arguments
```
Comment (kkoser) [inline src/cmd_line.h]: Why did we remove the brief/details only here?

**Comment by rboccabe** [comment]
Other than Kyle's, LGTM.

File: src/cmd_line.h (hunk 1 above)
Comment (alexmwu) [inline src/cmd_line.h]: my bad. will readd.

**Comment by rboccabe** [comment]
Pull then merge please.

**Pull request merged**